### DISCUSSION TITLE
Moving from VAST to tinyVAST : some questions

### DISCUSSION THREAD

**Discussion by MaxOlmos** [vast-lib/tinyVAST#39] [Q&A] [answered]
Hi Jim, 

I'm now switching from VAST to tinyVAST. Thank you very much for the body condition vignette. I am trying to apply the [tinyVAST](https://vast-lib.github.io/tinyVAST/articles/web_only/condition.html) vignette for conditions and density data (and compare to the VAST condtion vignette) and I am facing some issues.

1/ If I understood well alpha_j represent the year intercepts for each data type (condition and biomass) and the effect on log_length ?

2/ When running fit$opt$par (like in VAST) to show the parameters estimates (outer-optimizer, so it shows fixed effects only ?), there is no alpha_j. Does it mean that the alpha_j are random ? 

3/ Indeed I do not understand control=tinyVASTcontrol( trace=0, verbose=TRUE, profile="alpha_j" ). Do you define here your random effects ? Is it similar to random= c(“alpha_j”) in TMB but it does not make sense to me because alpha_j are different kind of parameters (intercepts and effect on log_length). 

4/ I looked at the tinyVAST function. If I want all the parameters estimates, is it OK if I run fit$internal$parlist ? if I want the values of the GMRF I have to use fit$internal$parlist$epsilon_stc and fit$internal$parlist$omega_sc ?

5/ Finally, I understand that the parameters theta are linked to sem (spatial) and beta to dsem (spatiotemporal). In our condition case study, theta[1] for the biomass, theta[2] for the condition and theta[3] for the biomass on the condition (idem for beta). But what is the nature of these parameters? I wasn't able to identify and name the parameters you presented in your [tinyVAST](https://arxiv.org/ftp/arxiv/papers/2401/2401.10193.pdf) article from equation (1)-(3)? Theta[3] is a fixed effect representing the effect of spatial variation of density on spatial variation on condition ? But what about theta[1] and theta[2] ? I would have been very interested in seeing the mathematical equations to make the link between the package and the mathematical equations.

Sorry for all the questions, but I'm trying to understand your package really well because it can be very useful for solving a lot of ecological problems.

Thank you very much.

Maxime

**Comment by James-Thorson-NOAA** [comment, marked as answer]
3) You could run with profile = "" to get the behavior you're expecting in Qs 2-3 ... using profile = "alpha_j" speeds up convergence but moves these parameters to the inner-optimizer without treating them as random effects.  

1)  correct, and you can run `summary(fit, "dsem")` and `summary(fit, "sem")` to see those DSEM and SEM parameters in a more interpretable presentation.

2)  See response to (3) 

4)  correct

5)  See response to (1) for better presentation of DSEM/SEM parameters, and `?make_dsem_ram`, plus the [arXiv](https://arxiv.org/abs/2401.10193) and [MEE](https://besjournals.onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.14289) papers (I'm guessing what you want is in `?make_dsem_ram`).  Do you have any specific suggestions for how I should extend the reference docs to make those topics more clear?

**Comment by James-Thorson-NOAA** [comment]
After thinking some more, I added a "seealso" section to the `?tinyVAST` reference docs with those links I mentioned in a new dev branch

**Reply by MaxOlmos** [reply]
Thank you very much Jim. The ?tinyVAST::make_dsem_ram is very useful to better understand the parameters ! Thanks !
